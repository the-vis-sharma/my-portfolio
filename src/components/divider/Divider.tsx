import { Box } from "@mui/material";
import React from "react";
import DarkVariant from "./DarkVariant";
import LightVariant from "./LightVariant";

interface DividerProps {
  width: string;
  variant: string;
}

const Divider: React.FunctionComponent<DividerProps> = (props) => {
  const { width, variant } = props;
  return (
    <Box
      sx={{ display: "flex", justifyContent: "center", alignItems: "center" }}
    >
      <Box sx={{ width }}>
        {variant === "light" ? <LightVariant /> : <DarkVariant />}
      </Box>
    </Box>
  );
};

export default Divider;
