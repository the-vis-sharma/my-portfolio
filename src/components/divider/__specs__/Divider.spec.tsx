import { render } from "@testing-library/react";
import Divider from "../Divider";

it("verify snapshot for divider in light mode", () => {
  const { asFragment } = render(<Divider variant="light" width="60%" />);
  expect(asFragment()).toMatchSnapshot();
});

it("verify snapshot for divider in dark mode", () => {
  const { asFragment } = render(<Divider variant="dark" width="80%" />);
  expect(asFragment()).toMatchSnapshot();
});
