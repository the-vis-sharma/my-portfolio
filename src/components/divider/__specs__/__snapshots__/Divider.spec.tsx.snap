// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify snapshot for divider in dark mode 1`] = `
<DocumentFragment>
  .c0 {
  height: 0.5px;
  background: #000;
  background: -webkit-linear-gradient( left, rgb(255,255,255) 0%, rgb(50,50,50) 50%, rgb(255,255,255) 100% );
}

.c0:before {
  position: relative;
  float: left;
  background: linear-gradient( to right, rgba(200,200,200,0), rgba(200,200,200,1) 50% );
  margin-top: 2px;
  width: 50%;
  height: 0.5px;
  content: "";
}

.c0:after {
  position: relative;
  float: right;
  background: linear-gradient( to left, rgba(100,200,200,0), rgba(200,200,200,1) 50% );
  margin-top: 2px;
  width: 50%;
  height: 0.5px;
  content: "";
}

<div
    class="MuiBox-root css-1n2mv2k"
  >
    <div
      class="MuiBox-root css-1x97c6v"
    >
      <div
        class="c0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`verify snapshot for divider in light mode 1`] = `
<DocumentFragment>
  .c0 {
  height: 0.5px;
  background: #fff;
  background: -webkit-linear-gradient( left, rgb(90,90,90) 0%, rgb(255,255,255) 50%, rgb(90,90,90) 100% );
}

.c0:before {
  position: relative;
  float: left;
  background: linear-gradient( to right, rgba(100,100,100,0), rgba(100,100,100,1) 50% );
  margin-top: 2px;
  width: 50%;
  height: 0.5px;
  content: "";
}

.c0:after {
  position: relative;
  float: right;
  background: linear-gradient( to left, rgba(100,100,100,0), rgba(100,100,100,1) 50% );
  margin-top: 2px;
  width: 50%;
  height: 0.5px;
  content: "";
}

<div
    class="MuiBox-root css-1n2mv2k"
  >
    <div
      class="MuiBox-root css-a6rva7"
    >
      <div
        class="c0"
      />
    </div>
  </div>
</DocumentFragment>
`;
